BitwardenShared/Core/Platform/Services/ConfigService.swift

```diff
@@ -64,6 +64,17 @@ protocol ConfigService {
         isPreAuth: Bool
     ) async -> String?
 
+    /// Retrieves a boolean feature flag from the passed  `config` argument. This is a synchronous operation
+    /// to be used when we already have the `ServerConfig` and we don't need to fetch it as in the async version
+    /// of this function.
+    ///
+    /// - Parameters:
+    ///   - flag: The feature flag to retrieve.
+    ///   - defaultValue: The default value to use if the flag is not in the server configuration.
+    ///   - config: The `ServerConfig` to get the value from.
+    /// - Returns: The value for the feature flag.
+    func getFeatureFlag(_ flag: FeatureFlag, defaultValue: Bool, from config: ServerConfig?) -> Bool
```

Comment:
Yeah I was really in doubt whether to add it or not. My line of thought was to avoid async calling on this so we can use it in sync contexts when we have the `ServerConfig` passed from the caller; as it's in this specific case. Also it saves going to disk to get the `ServerConfig` from user defaults.
However, I see the cons as well on having to maintain this additional code as well.
What are your thoughts? @matt-livefront and what about you @KatherineInCode ?

Reply:
I don't know how much I like expanding the API surface of the `ConfigService` like this. On the whole, I feel like a `ServerConfig` should remain a largely internal object, and shouldn't be passed in from something external—especially as at that point we start having to make sure we're not crossing the streams (as it were) by passing in an out-of-date configuration or the wrong configuration. That we serve it up at all in the function for "please make sure we're updated" is a concession to needing to pass information from it to the SDK, in my estimation.

The internals of the `ConfigService` also feel like it's getting to the point where I need a flowchart to keep track of the logic of how it all works in terms of what calls what 😅 and while I understand there's certainly complexity in that decision tree—complected by the fact that we need to triplicate most things for bool/int/string—I still would like to keep it as simple as we can, if that makes sense?

Reply:
Yeah for sure, I think it'll work in async manner as well so I'll make the change so that it remains as simple as possible. Thank you both for the feedback!